app/src/main/java/org/kiwix/kiwixmobile/local_file_transfer/LocalFileTransferFragment.java

```diff
@@ -101,25 +106,32 @@
 
   private WifiPeerListAdapter wifiPeerListAdapter;
 
-  @Override
-  protected void onCreate(@Nullable Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    setContentView(R.layout.activity_local_file_transfer);
+  @Nullable @Override public View onCreateView(@NonNull LayoutInflater inflater,
+    @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    View root = inflater.inflate(R.layout.fragment_local_file_transfer, container, false);
+    ButterKnife.bind(this, root);
+    return root;
+  }
+
+  @Override public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+    CoreMainActivity activity = (CoreMainActivity) requireActivity();
+    setHasOptionsMenu(true);
+    wifiDirectManager.setCallbacks(this);
     /*
      * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
      * - On the sender device, this activity is started from the app chooser post selection
      * of files to share in the Library
      * - On the receiver device, the activity is started directly from within the 'Get Content'
      * activity, without any file Uris
      * */
-    Intent filesIntent = getIntent();
+    Intent filesIntent = activity.getIntent();
```

Comment:
I don't think this will work, do you know how to test the local file transfer?

Reply:
No, I have two phones, that could work?

Reply:
multi select files and then.... share and choose kiwix? It has been a while since I tested this myself. The list of files will probably want to be args passed to it I imagine